code/game/objects/items/religion.dm

```diff
@@ -1,27 +1,126 @@
 /obj/item/banner
 	name = "banner"
+	desc = "A banner with Nanotrasen's logo on it."
 	icon = 'icons/obj/items_and_weapons.dmi'
 	icon_state = "banner"
 	item_state = "banner"
+	force = 8
+	attack_verb = list("forcefully inspired", "violently encouraged", "relentlessly galvanized")
 	lefthand_file = 'icons/mob/inhands/equipment/banners_lefthand.dmi'
 	righthand_file = 'icons/mob/inhands/equipment/banners_righthand.dmi'
-	desc = "A banner with Nanotrasen's logo on it."
-	var/moralecooldown = 0
-	var/moralewait = 600
+	var/morale_time = 0
+	var/morale_cooldown = 600 //How many deciseconds between uses
+	var/list/job_loyalties //Mobs with any of these assigned roles will be inspired
+	var/list/role_loyalties //Mobs with any of these special roles will be inspired
+	var/warcry
 
 /obj/item/banner/attack_self(mob/living/carbon/human/user)
-	if(moralecooldown + moralewait > world.time)
+	if(morale_time > world.time)
+		to_chat(user, "<span class='warning'>You aren't feeling inspired enough to flourish [src] again yet.")
 		return
-	to_chat(user, "<span class='notice'>You increase the morale of your fellows!</span>")
-	moralecooldown = world.time
-
-	for(var/mob/living/carbon/human/H in range(4,get_turf(src)))
-		to_chat(H, "<span class='notice'>Your morale is increased by [user]'s banner!</span>")
-		H.adjustBruteLoss(-15)
-		H.adjustFireLoss(-15)
-		H.AdjustStun(-40)
-		H.AdjustKnockdown(-40)
-		H.AdjustUnconscious(-40)
+	user.visible_message("<span class='big notice'>[user] flourishes [src]!</span>", \
+	"<span class='notice'>You raise [src] skywards, inspiring your allies!</span>")
+	playsound(user, "rustle", 100, FALSE)
+	if(warcry)
+		user.say("[warcry]!!")
+	user.transform *= 1.2
+	animate(user, transform = initial(user.transform), time = 10)
+	morale_time = world.time + morale_cooldown
+
+	var/list/inspired = list()
+	var/has_job_loyalties = LAZYLEN(job_loyalties)
+	var/has_role_loyalties = LAZYLEN(role_loyalties)
+	for(var/mob/living/carbon/human/H in range(4, get_turf(src)))
+		if(H.stat == DEAD)
+			continue
+		if(H.mind && (has_job_loyalties || has_role_loyalties))
+			if(has_job_loyalties && H.mind.assigned_role in job_loyalties)
+				inspired += H
+			else if(has_role_loyalties && H.mind.special_role in role_loyalties)
+				inspired += H
+		else if(check_inspiration(H))
+			inspired += H
+	inspired |= user //The user is always inspired regardless of loyalty
+
+	for(var/V in inspired)
+		var/mob/living/carbon/human/H = V
+		if(H != user)
+			to_chat(H, "<span class='notice'>Your confidence surges as [user] flourishes their [name]!</span>")
```

Comment:
their [src]

Reply:
That would lead to "their the meditopia banner"

Reply:
@ChangelingRain added a bunch of pronoun procs, please use them https://github.com/tgstation/tgstation/blob/551ede6d35bc955fa83979049441321f46ab9653/code/__HELPERS/pronouns.dm